src/test/java/com/epam/hw4/steps/AssertSteps.java

```diff
@@ -0,0 +1,136 @@
+
+    @Step("Assert Browser title")
+    public void browserTitle() {
+        softAssert.assertEquals(driver.getTitle(), "Home Page");
+        softAssert.assertAll();
+    }
+
+    @Step("assertion login")
+    public void assertLogin() {
+        String userName = BaseTest.mainPage.getUserName().getText();
+        softAssert.assertEquals(userName, "ROMAN IOVLEV");
+        softAssert.assertAll();
+    }
+
+    @Step("Items in header")
+    public void headerItems() {
+        int headerCount = BaseTest.mainPage.headerItems.size();
+        assertThat(headerCount).isEqualTo(4);
+        List<String> headerItems = BaseTest.mainPage.headerItems
+                .stream().map(WebElement::getText).collect(Collectors.toList());
+        assertThat(headerItems).containsExactlyElementsOf(itemHeader);
+        BaseTest.mainPage.headerItems.forEach(e -> assertTrue(e.isDisplayed()));
+    }
+
+    @Step("Images on the Index Page")
+    public void imagesIndexPage() {
+        int iconsCount = BaseTest.mainPage.iconsItems.size();
+        softAssert.assertEquals(iconsCount, 4);
+        BaseTest.mainPage.iconsItems.forEach(e -> assertTrue(e.isDisplayed()));
+        softAssert.assertAll();
+    }
+
+    @Step("Text under images on the Index Page")
+    public void textUnderImagesPage() {
+        int iconsTextCount = BaseTest.mainPage.iconsTextItems.size();
+        assertThat(iconsTextCount).isEqualTo(4);
+        List<String> textUnderIcons = BaseTest.mainPage.iconsTextItems
+                .stream().map(WebElement::getText).collect(Collectors.toList());
+        assertThat(textUnderIcons).containsExactlyElementsOf(iconsText);
+    }
+
+    @Step("Text under images on the Index Page (fail)")
+    public void textUnderImagesPageFail() {
+        int iconsTextCount = BaseTest.mainPage.iconsTextItems.size();
+        assertThat(iconsTextCount).isEqualTo(4);
+        List<String> textUnderIcons = BaseTest.mainPage.iconsTextItems
+                .stream().map(WebElement::getText).collect(Collectors.toList());
+        assertThat(textUnderIcons).containsExactlyElementsOf(iconsTextFail);
+    }
+
+    @Step("frame is displayed")
+    public void frame() {
+        softAssert.assertTrue(BaseTest.mainPage.frame.isDisplayed());
+        softAssert.assertAll();
+    }
+
+    @Step("Frame button is displayed")
+    public void frameButton() {
+        softAssert.assertTrue(BaseTest.mainPage.frameButton.isDisplayed());
+        softAssert.assertAll();
+    }
+
+    @Step("Items in the Left Section")
+    public void itemsLeftSection() {
+        int leftMenuCount = BaseTest.mainPage.leftMenu.size();
+        assertThat(leftMenuCount).isEqualTo(5);
+        List<String> leftMenuItems = BaseTest.mainPage.leftMenu
+                .stream().map(WebElement::getText).collect(Collectors.toList());
+        assertThat(leftMenuItems).containsExactlyElementsOf(itemLeftMenu);
+    }
+
+    @Step("Checkbox is selected")
+    public void checkboxIsSelected(String element) {
```

Comment:
instead of making the same logic twice (code duplicate) in assertion and action steps just need to get checkboxs, radiobutton or dropdown element one time and work with it further.